fastNLP/core/metrics/bleu.py

```diff
@@ -25,84 +22,99 @@ def tokenizer_fn(sentence: str) -> Sequence[str]:
         return np.array(0.)
     return np.array(np.exp(1 - references_len / pred_len))
 
 
-class Bleu(Metric):
+class BLEU(Metric):
+    """计算 bleu 的 metric 。
+
+    :param n_gram: Gram的范围是[1,4]
+    :param smooth: 是否选择smoothing计算
+    :param ngram_weights: 用来控制各个i-gram所计算结果的权重，sum(nrgam_weights)的结果为 1
+    :param backend: 目前支持五种类型的backend, ``['auto', 'torch', 'paddle', 'jittor', 'oneflow']``。
+        其中 ``'auto'`` 表示根据实际调用。
+        :meth:`update` 函数时传入的参数决定具体的 backend ，一般情况下直接使用 ``'auto'`` 即可。
+    :param aggregate_when_get_metric: 在计算 metric 的时候是否自动将各个进程上的相同的 element 的数字
+        聚合后再得到 metric，当 ``backend`` 不支持分布式时，该参数无意义。
+        如果为 ``None`` ，将在 :class:`~fastNLP.core.controllers.Evaluator`
+        中根据 ``sampler`` 是否使用分布式进行自动设置。
     """
-      计算 bleu 的 metric 。
-      :param n_gram: Gram的范围是[1,4]
-      :param smooth: 是否选择smoothing计算
-      :param ngram_weights: 用来控制各个i-gram所计算结果的权重，sum(nrgam_weights)的结果为 1
-      :param backend: 目前支持五种类型的backend, ``['auto', 'torch', 'paddle', 'jittor', 'oneflow']``。其中 ``'auto'`` 表示根据实际调用
-          :meth:`update` 函数时传入的参数决定具体的 backend ，一般情况下直接使用 ``'auto'`` 即可。
-      :param aggregate_when_get_metric: 在计算 metric 的时候是否自动将各个进程上的相同的 element 的数字聚合后再得到 metric，
-          当 ``backend`` 不支持分布式时，该参数无意义。如果为 ``None`` ，将在 :class:`~fastNLP.core.controllers.Evaluator`
-          中根据 ``sampler`` 是否使用分布式进行自动设置。
-      """
 
     def __init__(
-            self,
-            n_gram: int = 4,
-            smooth: bool = False,
-            ngram_weights: Optional[Sequence[float]] = None,
-            backend: Union[str, Backend, None] = 'auto',
-            aggregate_when_get_metric: bool = None,
-            **kwargs: Any,
+        self,
+        n_gram: int = 4,
+        smooth: bool = False,
+        ngram_weights: Optional[Sequence[float]] = None,
+        backend: Union[str, Backend, None] = 'auto',
+        aggregate_when_get_metric: bool = None,
+        **kwargs: Any,
     ):
-        super(Bleu, self).__init__(backend=backend, aggregate_when_get_metric=aggregate_when_get_metric)
+        super().__init__(backend=backend,
+                         aggregate_when_get_metric=aggregate_when_get_metric)
         self.n_gram = n_gram
         self.smooth = smooth
         if ngram_weights is not None and len(ngram_weights) != n_gram:
-            raise ValueError(f"List of weights has different weights than `n_gram`: {len(ngram_weights)} != {n_gram}")
-        self.ngram_weights = ngram_weights if ngram_weights is not None else [1.0 / n_gram] * n_gram
-
-        self.register_element(name="pred_len", value=0, aggregate_method='sum', backend=backend)
-        self.register_element(name="references_len", value=0, aggregate_method='sum', backend=backend)
-        self.register_element(name="precision_matches", value=[0 for _ in range(self.n_gram)], aggregate_method='sum',
+            raise ValueError(
+                f'List of weights has different weights than `n_gram`: {len(ngram_weights)} != {n_gram}'
```

Comment:
raise ValueError(f'The number of weights in `weights` is different from `n_gram`: {len(ngram_weights)} != {n_gram}'